evcache-core/src/main/java/net/spy/memcached/protocol/ascii/MetaArithmeticOperationImpl.java

```diff
@@ -0,0 +1,116 @@
+package net.spy.memcached.protocol.ascii;
+
+import net.spy.memcached.KeyUtil;
+import net.spy.memcached.ops.Mutator;
+import net.spy.memcached.ops.MutatorOperation;
+import net.spy.memcached.ops.OperationCallback;
+import net.spy.memcached.ops.OperationStatus;
+import net.spy.memcached.ops.OperationState;
+import net.spy.memcached.ops.StatusCode;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.Collections;
+
+public class MetaArithmeticOperationImpl extends EVCacheOperationImpl implements
+        MutatorOperation {
+
+    private static final Logger log = LoggerFactory.getLogger(MetaArithmeticOperationImpl.class);
+    private static final OperationStatus NOT_FOUND = new OperationStatus(false,
+            "NOT_FOUND", StatusCode.ERR_NOT_FOUND);
+
+    // TODO : Move to a Builder as we expand this to support better isolation guarantees
+    // Request
+    private static final String META_ARITHMETIC_OP = "ma";
+    private static final String AUTO_CREATE = "N%d";
+    private static final String MUTATOR_MODE ="M%c";
+    private static final char INCR = '+';
+    private static final char DECR = '-';
+    private static final String DEFAULT = "J%d";
+    private static final String DELTA = "D%d";
+    private static final char FLAG_VALUE = 'v';
+    // Response
+    private static final String VALUE_RETURN = "VA";
+
+
+    private final Mutator mutator;
+    private final String key;
+    private final long amount;
+    private final long def;
+    private final int exp;
+    private boolean readingValue;
+    public static final int OVERHEAD = 32;
+
+    public MetaArithmeticOperationImpl(Mutator m, String k, long amt, long defaultVal,
+                                       int expiry, OperationCallback c) {
+        super(c);
+        mutator = m;
+        key = k;
+        amount = amt;
+        def = defaultVal;
+        exp = expiry;
+        readingValue = false;
+    }
+
+    @Override
+    public void handleLine(String line) {
```

Comment:
given that core of the logic in the class is protocol implementation, I am not reviewing the logic itself.